velox/functions/prestosql/Sequence.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include "velox/expression/VectorFunction.h"
+#include "velox/vector/ConstantVector.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+template <typename TInput1, typename TInput2>
```

Comment:
Looks like TInput1 and TInput2 are always the same. Let's change to have just one template parameter T.

Reply:
As of now, they're the same, but I assumed in the future, we might need the function to also support other scenarios like sequence(int, long)? 
If that's the case, we should remain this way?

Reply:
@mbasmanova 
@kewang1024

Indeed, seems start/stop and step can be of different type:

Generate a sequence of dates from start to stop, incrementing by step. The type of step can be either INTERVAL DAY TO SECOND or INTERVAL YEAR TO MONTH.

Reply:
@kewang1024 I don't see such signatures in Presto. Let's use a single template parameter and change that if/when we need to. I also don't see Presto allowing 'integer' as input parameter. Only 'bigint' seems to be allowed. This suggests that we don't need a template parameter at all.

```
presto> show functions like 'sequence';
 Function |   Return Type    |                Argument Types                | Function Type | Deterministic |                  Description                   | Va>
----------+------------------+----------------------------------------------+---------------+---------------+------------------------------------------------+--->
 sequence | array(bigint)    | bigint, bigint                               | scalar        | true          |                                                | fa>
 sequence | array(bigint)    | bigint, bigint, bigint                       | scalar        | true          | Sequence function to generate synthetic arrays | fa>
 sequence | array(date)      | date, date                                   | scalar        | true          |                                                | fa>
 sequence | array(date)      | date, date, interval day to second           | scalar        | true          |                                                | fa>
 sequence | array(date)      | date, date, interval year to month           | scalar        | true          |                                                | fa>
 sequence | array(timestamp) | timestamp, timestamp, interval day to second | scalar        | true          |                                                | fa>
 sequence | array(timestamp) | timestamp, timestamp, interval year to month | scalar        | true          |                                                | fa>
```

Reply:
@spershin 

> seems start/stop and step can be of different type:

This is the case, but it doesn't look like template the class would be sufficient to support these. I suggest we either remove template parameter or add support for all the signatures.